doc/source/ray-core/fault_tolerance/actors.rst

```diff
@@ -135,6 +135,29 @@ original handle to the actor.
 
 If ``max_restarts`` is set, you can also allow Ray to automatically restart the actor by passing ``no_restart=False`` to ``ray.kill``.
 
+Unavailable actors
+----------------------
+
+When an actor can't accept method calls, a ``ray.get`` on the method's returned object reference may raise
+``ActorUnavailableError``. This exception indicates the actor isn't accessible at the
+moment, but may recover after waiting and retrying. Typical cases include:
+
+- The actor is restarting. For example, it's waiting for resources or running the class constructor during the restart.
```

Comment:
Does actor restarting use up task retries? If yes, is that new behavior? I think this behavior would be a bit odd, I would've expected task retries to get queued until the actor is done restarting.

If this is not the behavior, I think we can explain that case a bit more, i.e. whether retries would get consumed or not.

Reply:
Yes, actor restarting uses task retries. This is because when a task gets a ActorUnavailableError, that task is already failed, so a retry must be consumed in order to have a second attempt. If we don't eat a task retry count, we may end found task retry count < number of tasks issued, which could be weird.

The behavior is already existing in https://github.com/ray-project/ray/issues/22818 (fast fail tasks when actor restarting). I can add a line in the doc about how this eats retries.